### PR TITLE
Thunder schema sh refactoring add interfaces

### PR DESCRIPTION
**Pull request by shagel7984**
Thunder schema sh refactoring add interfaces

### CODE REVIEW

File: src/Plugin/GraphQL/Schema/ThunderSchema.php [hunk 1]
```diff
@@ -61,43 +61,61 @@ public function getResolverRegistry() {
...
+   * Function addInterfaces.
    *
    */
   protected function addInterfaces () {
     $this->registry->addTypeResolver('ContentType', function ($value) {
+      $bundle = $value->bundle();
+
       if ($value instanceof ContentEntityInterface) {
-        return ucfirst($value->bundle());
+        return $this->mapBundleToSchemaName();
```
Comment (dbosen) [inline src/Plugin/GraphQL/Schema/ThunderSchema.php]: The bundle parameter is missing

File: src/Plugin/GraphQL/Schema/ThunderSchema.php [hunk 2]
```diff
@@ -61,43 +61,61 @@ public function getResolverRegistry() {
...
       }
-      throw new Error('Could not resolve content type.');
+      throw new Error('Could not resolve content type.' . $bundle);
```
Comment (dbosen) [inline src/Plugin/GraphQL/Schema/ThunderSchema.php]: Please add space after the colon for better readability

File: src/Plugin/GraphQL/Schema/ThunderSchema.php [hunk 3]
```diff
@@ -61,43 +61,61 @@ public function getResolverRegistry() {
...
+        return 'Paragraph' . $this->mapBundleToSchemaName($bundle);
       }
 
       if($value instanceof MediaInterface) {
-        return ucfirst($value->bundle());
+        return $this->mapBundleToSchemaName($bundle);
       }
 
-      throw new Error('Could not resolve element type.');
+      throw new Error('Could not resolve element type.' . $bundle);
```
Comment (dbosen) [inline src/Plugin/GraphQL/Schema/ThunderSchema.php]: Space missing as well

File: src/Plugin/GraphQL/Schema/ThunderSchema.php [hunk 4]
```diff
@@ -61,43 +61,61 @@ public function getResolverRegistry() {
...
     });
 
   }
 
+  /**
+   * Takes the bundle name and returns the schema name.
+   * @param $bundle_name
+   *   The bundle name.
+   *
+   * return bundle_name
```
Comment (dbosen) [inline src/Plugin/GraphQL/Schema/ThunderSchema.php]: that should be` * @return string` instead

File: src/Plugin/GraphQL/Schema/ThunderSchema.php [hunk 5]
```diff
@@ -61,43 +61,61 @@ public function getResolverRegistry() {
...
+   *   Returns the mapped bundle name.
+   */
+  protected function mapBundleToSchemaName($bundle_name) {
```
Comment (dbosen) [inline src/Plugin/GraphQL/Schema/ThunderSchema.php:107]: I think this whole Function can be reduced to 
```

protected function mapBundleToSchemaName($bundle_name) {
  return str_replace('_', '', ucwords($bundle_name, '_'));
}

```

**Review by dbosen — CHANGES_REQUESTED** [review]
Some requests below

File: src/Plugin/GraphQL/Schema/ThunderSchema.php (hunk 5 above)
Comment (shagel7984) [reply src/Plugin/GraphQL/Schema/ThunderSchema.php:107]: very nice

File: src/Plugin/GraphQL/Schema/ThunderSchema.php (hunk 4 above)
Comment (shagel7984) [reply src/Plugin/GraphQL/Schema/ThunderSchema.php]: ok

File: src/Plugin/GraphQL/Schema/ThunderSchema.php (hunk 3 above)
Comment (shagel7984) [reply src/Plugin/GraphQL/Schema/ThunderSchema.php]: ok

File: src/Plugin/GraphQL/Schema/ThunderSchema.php (hunk 2 above)
Comment (shagel7984) [reply src/Plugin/GraphQL/Schema/ThunderSchema.php]: ok

File: src/Plugin/GraphQL/Schema/ThunderSchema.php (hunk 1 above)
Comment (shagel7984) [reply src/Plugin/GraphQL/Schema/ThunderSchema.php]: right thanks

**Review by dbosen — APPROVED** [review]
looks good!

**Pull request merged**